cli-plugins/plugin/plugin.go

```diff
@@ -27,21 +27,15 @@ import (
 var PersistentPreRunE func(*cobra.Command, []string) error
 
 // RunPlugin executes the specified plugin command
-func RunPlugin(dockerCli *command.DockerCli, plugin *cobra.Command, meta manager.Metadata) error {
-	tcmd := newPluginCommand(dockerCli, plugin, meta)
+func RunPlugin(ctx context.Context, dockerCli *command.DockerCli, plugin *cobra.Command, meta manager.Metadata) error {
```

Comment:
Are we fine with a breaking change in the public function signature that's used by every CLI plugin?

Reply:
Yeah this will break plugins on the next version. This is a tricky one since **not** passing a context could cause problems further down the line (like graceful cleanup). And I'd like to move away from using `context.TODO()` or `context.Background()` instead in these kind of places. 

The next best thing would be to use the `dockerCli.baseCtx` but this is an anti-pattern in terms of context and we should actually look to remove this from the `command.DockerCli` struct.

Reply:
removed this - see new comments on the PR for more details.